recipe/build.sh

```diff
@@ -1,60 +1,77 @@
-cp $BUILD_PREFIX/share/gnuconfig/config.* .
+# cp $BUILD_PREFIX/share/gnuconfig/config.* .
 
 set -ex # Abort on error.
 
 # recommended in https://gitter.im/conda-forge/conda-forge.github.io?at=5c40da7f95e17b45256960ce
-find ${PREFIX}/lib -name '*.la' -delete
+# find ${PREFIX}/lib -name '*.la' -delete
 
 # Force python bindings to not be built.
-unset PYTHON
+# unset PYTHON
 
-export CPPFLAGS="${CPPFLAGS} -I${PREFIX}/include"
-# Remove -std=c++XX from build ${CXXFLAGS}
-CXXFLAGS=$(echo "${CXXFLAGS}" | sed -E 's@-std=c\+\+[^ ]+@@g')
-export CXXFLAGS="${CXXFLAGS} -std=c++17"
+# export CPPFLAGS="${CPPFLAGS} -I${PREFIX}/include"
+# # Remove -std=c++XX from build ${CXXFLAGS}
+# CXXFLAGS=$(echo "${CXXFLAGS}" | sed -E 's@-std=c\+\+[^ ]+@@g')
+# export CXXFLAGS="${CXXFLAGS} -std=c++17"
 # also allow newer symbols (https://conda-forge.org/docs/maintainer/knowledge_base.html#newer-c-features-with-old-sdk)
 export CXXFLAGS="${CXXFLAGS} -D_LIBCPP_DISABLE_AVAILABILITY"
 
-(bash configure --prefix=${PREFIX} \
-               --host=${HOST} \
-               --with-blosc=${PREFIX} \
-               --with-curl \
-               --with-dods-root=${PREFIX} \
-               --with-expat=${PREFIX} \
-               --with-freexl=${PREFIX} \
-               --with-geos=${PREFIX}/bin/geos-config \
-               --with-geotiff=${PREFIX} \
-               --with-hdf4=${PREFIX} \
-               --with-cfitsio=${PREFIX} \
-               --with-hdf5=${PREFIX} \
-               --with-tiledb=${PREFIX} \
-               --with-jpeg=${PREFIX} \
-               --with-kea=${PREFIX}/bin/kea-config \
-               --with-libiconv-prefix=${PREFIX} \
-               --with-libjson-c=${PREFIX} \
-               --with-libkml=${PREFIX} \
-               --with-liblzma=yes \
-               --with-libtiff=${PREFIX} \
-               --with-libz=${PREFIX} \
-               --with-netcdf=${PREFIX} \
-               --with-openjpeg=${PREFIX} \
-               --with-pcre \
-               --with-pg=yes \
-               --with-png=${PREFIX} \
-               --with-poppler=${PREFIX} \
-               --with-spatialite=${PREFIX} \
-               --with-sqlite3=${PREFIX} \
-               --with-proj=${PREFIX} \
-               --with-webp=${PREFIX} \
-               --with-xerces=${PREFIX} \
-               --with-xml2=yes \
-               --with-zstd=${PREFIX} \
-               --without-python \
-               --disable-static \
-               --verbose \
-               ${OPTS}) || (cat config.log; false)
-
-make -j $CPU_COUNT ${VERBOSE_AT}
+mkdir build
+cd build
 
+cmake -G "Unix Makefiles" \
+      ${CMAKE_ARGS} \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DCMAKE_PREFIX_PATH=$PREFIX \
+      -DCMAKE_INSTALL_PREFIX=$PREFIX \
+      -DCMAKE_INSTALL_LIBDIR=lib \
+      -DBUILD_SHARED_LIBS=ON \
+      -DBUILD_PYTHON_BINDINGS:BOOL=OFF \
+      -DBUILD_JAVA_BINDINGS:BOOL=OFF \
+      -DBUILD_CSHARP_BINDINGS:BOOL=OFF \
+      ${SRC_DIR}
```

Comment:
I can't thank everyone enough for taking this on!  My only wonder is being implicit vs explicit with the `cmake` command. Would it be better to explicitly include all the packages desired for cmake to attempt to build against?  I liked the verbosity of `--with-XXX` with the old configure...

Reply:
I see what you mean, but I think on balance I come down against explicit. Essentially, that could serve three purposes, first to document the included packages, second to guard against the removal of packages upstream, and third to guard against the addition of packages. The first is better served by @gillins addition of a dump of the supported formats to the tests, the second and third are somewhat served by the improved checks in [`run_test.py`](https://github.com/conda-forge/gdal-feedstock/blob/2972fd6abec521ed6f01f153207b8b90b6fc0636/recipe/run_test.py), but also might not be helped by being explicit here. This is because with Cmake, we are just setting variables here, which will not fail the run either way. If the package is removed, there will only be [a warning that variable x is not being used by this package](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=567669&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=841356e0-85bb-57d8-dbbc-852e683d1642&l=1538); if a new package is added but not mentioned here, it will still be autodetected, won't it?